x-pack/plugins/observability_solution/infra/public/components/asset_details/tabs/dashboards/dashboards.tsx

```diff
@@ -0,0 +1,203 @@
+import { ContextMenu } from './context_menu';
+import { useAssetDetailsUrlState } from '../../hooks/use_asset_details_url_state';
+
+export function Dashboards() {
+  const { dateRange } = useDatePickerContext();
+  const { asset } = useAssetDetailsRenderPropsContext();
+  const [dashboard, setDashboard] = useState<AwaitingDashboardAPI>();
+  const [customDashboards, setCustomDashboards] = useState<DashboardItemWithTitle[]>([]);
+  const [currentDashboard, setCurrentDashboard] = useState<DashboardItemWithTitle>();
+  const { data: allAvailableDashboards, status } = useDashboardFetcher();
+  const { metrics } = useDataViewsContext();
+  const [urlState, setUrlState] = useAssetDetailsUrlState();
+
+  const { dashboards, loading, reload } = useFetchCustomDashboards({ assetType: asset.type });
+
+  useEffect(() => {
+    const allAvailableDashboardsMap = new Map<string, DashboardItem>();
+    allAvailableDashboards.forEach((availableDashboard) => {
+      allAvailableDashboardsMap.set(availableDashboard.id, availableDashboard);
+    });
+    const filteredCustomDashboards =
+      dashboards?.reduce<DashboardItemWithTitle[]>(
+        (result: DashboardItemWithTitle[], customDashboard: InfraSavedCustomDashboard) => {
+          const matchedDashboard = allAvailableDashboardsMap.get(
+            customDashboard.dashboardSavedObjectId
+          );
+          if (matchedDashboard) {
+            result.push({
+              title: matchedDashboard.attributes.title,
+              ...customDashboard,
+            });
+          }
+          return result;
+        },
+        []
+      ) ?? [];
+    setCustomDashboards(filteredCustomDashboards);
+    // set a default dashboard if there is no selected dashboard
+    if (!urlState?.dashboardId) {
+      setUrlState({
+        dashboardId:
+          currentDashboard?.dashboardSavedObjectId ??
+          filteredCustomDashboards[0]?.dashboardSavedObjectId,
+      });
+    }
+  }, [
+    allAvailableDashboards,
+    currentDashboard?.dashboardSavedObjectId,
+    dashboards,
+    setUrlState,
+    urlState?.dashboardId,
+  ]);
+
+  const getCreationOptions = useCallback((): Promise<DashboardCreationOptions> => {
+    const getInitialInput = () => ({
+      viewMode: ViewMode.VIEW,
+      timeRange: { from: dateRange.from, to: dateRange.to },
+    });
+    return Promise.resolve<DashboardCreationOptions>({ getInitialInput });
+  }, [dateRange.from, dateRange.to]);
+
+  useEffect(() => {
+    if (!dashboard) return;
+    dashboard.updateInput({
+      filters:
+        metrics.dataView && currentDashboard?.dashboardFilterAssetIdEnabled
+          ? buildAssetIdFilter(asset.name, asset.type, metrics.dataView)
+          : [],
+      timeRange: { from: dateRange.from, to: dateRange.to },
+    });
+  }, [
+    metrics.dataView,
+    asset.name,
+    dashboard,
+    dateRange.from,
+    dateRange.to,
+    currentDashboard,
+    asset.type,
+  ]);
+
+  return (
+    <EuiPanel hasBorder>
```

Comment:
Would it be possible to simplify this nested ternary if?

Reply:
I moved the loading condition as an early return.